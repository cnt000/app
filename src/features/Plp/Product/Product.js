import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import LabelsContext from '../../Labels';
import { boolUnit, productAdded, listUnit } from '../../../state';
import { imageUrl } from '../../../utils/images';
import styles from './Product.module.css';
import PictureTagProduct from '../PictureTagProduct/PictureTagProduct';

const Product = ({ addToCartLink, i, image, name, price }) => {
  // outerWitdh / 2
  // se > 375 outerWidth / 3
  // se > 768 outerWidth / 4
  // se > 1024 outerWidth - 190 / 4
  const { addToCart } = useContext(LabelsContext);
  const itemLink = `/pdp/${addToCartLink.split('=')[2].split('&')[0]}`;

  function handleAdd(product) {
    listUnit.push(product);
    boolUnit.dispatch(true);
    productAdded.dispatch(true);
    setTimeout(() => boolUnit.dispatch(false), 2000);
  }

  return (
    <article key={`${name}_${i}`} className={styles.product}>
      <Link to={itemLink}>
        <div className={styles.imageContainer}>
          <PictureTagProduct
            className={styles.image}
            alt={name}
            image={imageUrl(image)}
          />
        </div>
        <h1 className={styles.name}>{name}</h1>
        <div className={styles.price}>â‚¬ {price}</div>
      </Link>
      <button
        className={styles.addToCart}
        onClick={(e) => handleAdd({ addToCartLink, i, image, name, price })}
      >
        {addToCart}
      </button>
    </article>
  );
};

Product.propTypes = {
  i: PropTypes.number,
  name: PropTypes.string,
  image: PropTypes.string,
  price: PropTypes.string,
  Link: PropTypes.object,
};

export default Product;
