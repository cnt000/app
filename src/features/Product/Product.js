import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import LabelsContext from '../Labels';
import { boolUnit, productAdded, listUnit } from '../../state';
import { getImageUrlCropped } from '../../utils/images';
import styles from './Product.module.css';

const imageUrl = (image) => image.split('/').pop().replace(' ', '%20');

const Product = ({ addToCartLink, i, image, name, price }) => {
  // outerWitdh / 2
  // se > 375 outerWidth / 3
  // se > 768 outerWidth / 4
  // se > 1024 outerWidth - 190 / 4
  const { addToCart } = useContext(LabelsContext);
  const itemLink = `/pdp/${addToCartLink.split('=')[2].split('&')[0]}`;

  const itemImage = (width) => getImageUrlCropped(imageUrl(image), width);

  function handleAdd(product) {
    listUnit.push(product);
    boolUnit.dispatch(true);
    productAdded.dispatch(true);
    setTimeout(() => boolUnit.dispatch(false), 2000);
  }

  return (
    <article key={`${name}_${i}`} className={styles.product}>
      <Link to={itemLink}>
        <div className={styles.imageContainer}>
          <img
            className={styles.image}
            src={itemImage(300)}
            srcSet={`${itemImage(200)} 200w,
              ${itemImage(256)} 256w,
              ${itemImage(450)} 450w,
              ${itemImage(650)} 580w,
              ${itemImage(650)} 660w`}
            alt={name}
            sizes="(max-width: 768px) 50vw, (max-width: 1024px) 33vw, 25vw"
            loading="lazy"
          />
        </div>
        <h1 className={styles.name}>{name}</h1>
        <div className={styles.price}>â‚¬ {price}</div>
      </Link>
      <button
        className={styles.addToCart}
        onClick={(e) => handleAdd({ addToCartLink, i, image, name, price })}
      >
        {addToCart}
      </button>
    </article>
  );
};

Product.propTypes = {
  i: PropTypes.number,
  name: PropTypes.string,
  image: PropTypes.string,
  price: PropTypes.string,
  Link: PropTypes.object,
};

export default Product;
