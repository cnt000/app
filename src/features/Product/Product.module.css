
.product {
  height: auto;
  width: calc(50% - var(--base-gap));
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  border: 1px solid var(--light-green);
  padding: calc(var(--base-gap) * 2) 0 0 0;
  margin: calc(var(--base-gap) / 2) 0;
  border-radius: var(--base-gap);
}

.name {
  margin: calc(var(--base-gap) * 3) auto;
}

.price {
  font-style: italic;
}

.addToCart {
  margin: var(--base-gap) auto calc(var(--base-gap) * 2) auto;
  background-color: var(--medium-green);
  color: white;
  padding: calc(var(--base-gap) * 3);
  border-radius: 3px;
  border: 1px solid white;
  cursor: pointer;
}

.addToCart:focus {
  border: 1px dotted var(--dark-green);
}

.addToCart:hover {
  filter: brightness(110%) saturate(140%);
}

.imageContainer {
  width: 96%;
  margin: 0 auto;
  position: relative;
  height: 0;
  overflow: hidden;
  padding-top: calc(1.33333333 * 100%);
  border-radius: var(--base-gap);
}

.image {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

@media only screen and (min-width: 768px) {
  .product {
    width: calc(33% - var(--base-gap));
  }
}

@media only screen and (min-width: 1024px) {
  .product {
    width: calc(25% - var(--base-gap));
    max-width: 400px;
    margin-bottom: 1%;
  }
}
