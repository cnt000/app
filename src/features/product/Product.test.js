import React from 'react';
import renderer from 'react-test-renderer';
import Product from './Product';

it('renders correctly', () => {
  const tree = renderer
    .create(
        <Product />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
