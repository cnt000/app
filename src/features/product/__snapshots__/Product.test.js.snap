// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  "domain: ",
  "http://localhost:8080",
  <section>
    <h1>
      name: 
    </h1>
    <img />
    <br />
    <div>
      size: 
    </div>
    <div>
      price: 
    </div>
    <br />
    <a>
      Add
    </a>
    <br />
  </section>,
]
`;
